frontend/lib/src/components/widgets/DataFrame/columns/cells/UriOverlayEditor.tsx

```diff
@@ -32,24 +29,9 @@ interface Props {
 // this is essentially just copying the UriOverlayEditor from glide's implementation: https://github.com/glideapps/glide-data-grid/blob/0ea52f371a5e2aaa8595aceefa40722d35410b1a/packages/core/src/data-grid-overlay-editor/private/uri-overlay-editor-style.tsx
 // we use it in LinkCell.tsx which is our custom version of the UriCell.
 const UriOverlayEditor: React.FunctionComponent<Props> = p => {
-  const {
-    uri,
-    onChange,
-    forceEditMode,
-    readonly,
-    validatedSelection,
-    preview,
-  } = p
+  const { uri, onChange, readonly, validatedSelection, preview } = p
 
-  const [editMode, setEditMode] = React.useState<boolean>(
-    !readonly && (!uri || forceEditMode)
```

Comment:
This just works out of the box with the changes since it will just always directly jump into editing if not read-only. 

https://github.com/streamlit/streamlit/assets/2852129/2bd70f90-0ac5-46e9-84c6-f2f52ad280e0